avocado/utils/cpu.py

```diff
@@ -97,28 +97,20 @@ def get_cpu_arch():
     """
     Work out which CPU architecture we're running on
     """
+    cpu_table = [('^cpu.*(RS64|POWER3|Broadband Engine)', 'power'),
+                 ('^cpu.*POWER4', 'power4'),
+                 ('^cpu.*POWER5', 'power5'),
+                 ('^cpu.*POWER6', 'power6'),
+                 ('^cpu.*POWER7', 'power7'),
+                 ('^cpu.*POWER8', 'power8'),
+                 ('^cpu.*PPC970', 'power970'),
```

Comment:
I agree that the fallback on `i386` is very very bad, and that we should change it.  I intentionally tried to keep the same behavior out of the public functions, but one subsequent commit can introduce the change in behavior.

BTW, this discussion is actually nice, because this module seems indeed in great need of some TLC, and some related work (guidelines for tests) made me look into it.

Now, a different matter: it baffles me when a NACK on one change (improvement?) is done on the grounds that a future (when? is there one now?) much more comprehensive set of changes will be proposed.

This should be true to **any** review: if there isn't something broken with what's being proposed, we should gently walk it inside the "master gates".  If there's something wrong, give the author a change to fix it.  Do not tell the author that he shouldn't care about working/sending those (minor?) proposals: it hurts the community contribution process.  A reviewer should not ACK or NACK based on serious or important the issue being addressed seems to be to himself/herself.  The author *already* took the time to write the code and propose the changes.  If a much improved and comprehensive proposal follows, even if that's a few days later, so be it.